pkg/apis/core/v1alpha1/types_common.go

```diff
@@ -117,11 +117,6 @@ const (
 	// ExternalGardenerName is the value in a Kubernetes core resources `.metadata.finalizers[]` array on which the
 	// Gardener will react when performing a delete request on a resource.
 	ExternalGardenerName = "gardener.cloud/gardener"
-	// ExternalGardenerNameDeprecated is the value in a Kubernetes core resources `.metadata.finalizers[]` array on which the
-	// Gardener will react when performing a delete request on a resource.
-	//
-	// Deprecated: Use `ExternalGardenerName` instead.
-	ExternalGardenerNameDeprecated = "garden.sapcloud.io/gardener"
```

Comment:
This finalizer might be part of some `Secret`s that exist since a long time - I don't see code that would handle/remove/replace them properly. WDYT?

Reply:
There was a migration code included in https://github.com/gardener/gardener/pull/1940. I also see a small number of Secrets that have this finalizer - I believe these are cases where user created the secret with the deprecated finalizer or finalizer was not removed from the Secret for some reason. I will have a look later today..

Reply:
I had a look and I can summarise in the following way:
 - I believe that most of the Secrets with that finalizer are created initially with the finalizer by the end user or the finalizer is re-added (for example with `kubectl apply`) again by the end users.
- There were several cases in which there was a Secret with this finalizer but no associated SecretBinding. I believe these Secrets are leftover from the issue resolved with https://github.com/gardener/gardener/pull/1927.
- There was a single case in which the namespace was gone but there was a Secret with that finalizer with the given namespace name. I recreated the ns to allow proper deletion for this Secret.

IMO we cannot do much about the cases above. As I wrote already above there was a migration login in #1940 (that was cleaned up in #2449) and I think I did the trick. However I cleaned up the finalizer from Secrets that match the 3 cases above. Let me know if you have further comments.

Reply:
Alright, thanks for checking so thoroughly.

/lgtm